great_expectations/data_context/store/store.py

```diff
@@ -194,6 +206,8 @@ def get(
             value = self._store_backend.get(self.key_to_tuple(key))
 
         if value:
+            if isinstance(value, list):
+                return [self.deserialize(v) for v in value]
```

Comment:
Bleh same note as above - don't love unions but this is necessary if we want to retrieve an entire collection (like we do when calling `context.datasources`)